src/lib/actions/RuleRemoveAction.ts

```diff
@@ -0,0 +1,105 @@
+import {Action} from "../ScannerCommand";
+import {Inputs, Rule} from "../../types";
+import {AnyJson} from "@salesforce/ts-types";
+import {stringArrayTypeGuard} from "../util/Utils";
+import {Logger, SfError} from "@salesforce/core";
+import {BundleName, getMessage} from "../../MessageCatalog";
+import {Controller} from "../../Controller";
+import {RuleFilter, SourcePackageFilter} from "../RuleFilter";
+import {Display} from "../Display";
+import {InputsResolver} from "../InputsResolver";
+
+/**
+ * The Action behind the "rule remove" command
+ */
+export class RuleRemoveAction implements Action {
+	private readonly logger: Logger;
+	private readonly display: Display;
+	private readonly inputsResolver: InputsResolver;
+
+	public constructor(logger: Logger, display: Display, inputsResolver: InputsResolver) {
+		this.logger = logger;
+		this.display = display;
+		this.inputsResolver = inputsResolver;
+	}
+
+	public validateInputs(inputs: Inputs): Promise<void> {
+		// --path '' results in different values depending on the OS. On Windows it is [], on *nix it is [""]
+		if (inputs.path && stringArrayTypeGuard(inputs.path) && (!inputs.path.length || inputs.path.includes(''))) {
+			throw new SfError(getMessage(BundleName.Remove, 'validations.pathCannotBeEmpty'));
+		}
+		return Promise.resolve();
+	}
+
+	public async run(inputs: Inputs): Promise<AnyJson> {
+		// Step 2: Pull out and process our flag.
```

Comment:
Either the numbers in these steps should be updated, or an additional comment should be added above it that says something like "Step 1 is validating input, which has already occurred by this point."

Reply:
OK will remove the numbering but keep the comments.

Reply:
Fixed - squashed into part 10 with rebase and then force push.